Sources/Swift/Integrations/SessionReplay/SentryOnDemandReplay.swift

```diff
@@ -0,0 +1,166 @@
+        let newHeight = newWidth / aspectRatio
+        
+        let newSize = CGSize(width: newWidth, height: newHeight)
+        
+        UIGraphicsBeginImageContextWithOptions(newSize, false, 1)
+        originalImage.draw(in: CGRect(x: 0, y: 0, width: newSize.width, height: newSize.height))
+        let resizedImage = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+        
+        return resizedImage
+    }
+    
+    func releaseFramesUntil(_ date: Date) {
+        _onDemandDispatchQueue.async {
+            while let first = self._frames.first, first.time < date {
+                self._frames.removeFirst()
+                try? FileManager.default.removeItem(at: URL(fileURLWithPath: first.imagePath))
+            }
+        }
+    }
+    
+    func createVideoWith(duration: TimeInterval, beginning: Date, outputFileURL: URL, completion: @escaping (SentryVideoInfo?, Error?) -> Void) throws {
+        let videoWriter = try AVAssetWriter(url: outputFileURL, fileType: .mov)
+        
+        let videoSettings = createVideoSettings()
+        
+        let videoWriterInput = AVAssetWriterInput(mediaType: .video, outputSettings: videoSettings)
+        let bufferAttributes: [String: Any] = [
+           String(kCVPixelBufferPixelFormatTypeKey): kCVPixelFormatType_32ARGB
+        ]
+        
+        let pixelBufferAdaptor = AVAssetWriterInputPixelBufferAdaptor(assetWriterInput: videoWriterInput, sourcePixelBufferAttributes: bufferAttributes)
+        
+        videoWriter.add(videoWriterInput)
+        videoWriter.startWriting()
+        videoWriter.startSession(atSourceTime: .zero)
+        
+        let end = beginning.addingTimeInterval(duration)
+        var frameCount = 0
+        var frames = [String]()
+        
+        var start = Date()
+        var actualEnd = Date()
+        
+        for frame in _frames {
+            if frame.time < beginning { continue } else if frame.time > end { break }
+            if frame.time < start { start = frame.time }
+            
+            actualEnd = frame.time
+            frames.append(frame.imagePath)
+        }
+        
+        if frames.isEmpty { return }
+        
+        _currentPixelBuffer = SentryPixelBuffer(size: CGSize(width: videoWidth, height: videoHeight))
+        
+        videoWriterInput.requestMediaDataWhenReady(on: _onDemandDispatchQueue) { [weak self] in
+            guard let self = self else { return }
+            
+            let imagePath = frames[frameCount]
+            if let image = UIImage(contentsOfFile: imagePath) {
+                let presentTime = CMTime(seconds: Double(frameCount), preferredTimescale: CMTimeScale(self.frameRate))
+                frameCount += 1
+                
+                if self._currentPixelBuffer?.append(image: image, pixelBufferAdapter: pixelBufferAdaptor, presentationTime: presentTime) != true {
+                    completion(nil, videoWriter.error)
```

Comment:
`guard` is more verbose when used just for an early return; usually, I prefer to use it only for optional binding.

And since the use of `guard` over `if` is a matter of preference, I won't be retroactively changing it in this PR, but I will be more aware of it from now on (despite not liking it).